crates/walrus-service/src/node/blob_sync.rs

```diff
@@ -228,6 +228,43 @@ impl BlobSyncHandler {
         Ok(count)
     }
 
+    /// Similar to `cancel_all_expired_syncs_and_mark_events_completed`, but for all blob syncs.
+    #[tracing::instrument(skip(self))]
+    pub async fn cancel_all_syncs_and_mark_events_completed(&self) -> anyhow::Result<usize> {
+        tracing::info!("cancelling all blob syncs");
+
+        let join_handles: Vec<_> = {
+            let mut in_progress_guard = self
+                .blob_syncs_in_progress
+                .lock()
+                .expect("should be able to acquire lock");
+            tracing::info!("acquired lock on the in-progress blob recoveries");
+
+            if cfg!(not(msim)) {
+                in_progress_guard
+                    .par_iter_mut()
+                    .filter_map(|(_, sync)| sync.cancel())
+                    .collect()
+            } else {
+                in_progress_guard
+                    .iter_mut()
+                    .filter_map(|(_, sync)| sync.cancel())
+                    .collect()
+            }
+        };
+        tracing::info!("released lock on the in-progress blob recoveries");
+
+        let count = join_handles.len();
+
+        try_join_all(join_handles)
+            .await?
+            .into_iter()
+            .for_each(CompletableHandle::mark_as_complete);
+
+        tracing::info!("cancelled {count} blob syncs");
+        Ok(count)
+    }
```

Comment:
Question: We have a separate function `cancel_all` and a large code overlap with `cancel_all_expired_syncs_and_mark_events_completed`. Could we consolidate these three functions somehow (maybe in a follow-up)?

Reply:
I actually have a follow-up PR almost ready, will create it and ask @halfprice for a review.